src/utils.ts

```diff
@@ -122,10 +127,23 @@ export function canonicalizeUrl(urlOrPath: string): string {
 
     urlOrPath = stripTrailingSlash(urlOrPath);
     urlOrPath = fixDriveLetterAndSlashes(urlOrPath);
+    if (!caseSensitivePaths) {
+        urlOrPath = normalizeIfFSIsCaseInsensitive(urlOrPath);
+    }
 
     return urlOrPath;
 }
 
+function normalizeIfFSIsCaseInsensitive(urlOrPath: string): string {
+    return isFilePath(urlOrPath)
+        ? urlOrPath.toLowerCase()
+        : urlOrPath;
+}
+
+export function isFilePath(candidate: string): boolean {
```

Comment:
This will only match Windows-style paths. If it won't match unix-style paths, can you give the method a more specific name? Does it need to be exported?

Reply:
Changed to isWindowsFilePath